### PR TITLE
feat(otelcol/exp/prom): add support for resource_to_telemetry_conversion

### LINKED ISSUE #5631: Support resource_to_telemetry_conversion in otelcol.exporter.prometheus
### Request

The OpenTelemetry [prometheus remotewrite exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusremotewriteexporter) supports a `resource_to_telemetry_conversion` config option which converts resource attributes to datapoint attributes. We need to add the same feature to `otelcol.exporter.prometheus`.

### Use case

`otelcol.processor.k8sattributes` attaches Kubernetes metadata as resource attributes. Those attributes need to be copied as data point attributes in order to not be lost when remote writing with `prometheus.remote_write`. This was recently [discussed](https://grafana.slack.com/archives/C01050C3D8F/p1697614064021499) in a community Slack. 

The only workaround so far is to use `otelcol.processor.transform`, but I believe this requires listing all attributes explicitly, which is inconvenient.

### PR DESCRIPTION
**Pull request by hainenber**
#### PR Description

Support conversion of OTEL's resource attributes to Prometheus-compatible datapoint attributes. Strongly inspired by [OTEL's contributor implementation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a80edda1860b50fde9a2e7d6d9df9152cc49e87c/pkg/resourcetotelemetry/resource_to_telemetry.go#L20)

#### Which issue(s) this PR fixes

Closes #5631 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Comment by rfratto** [comment]
Thanks @hainenber! 

@ptodev Can you review this one?

File: CHANGELOG.md [hunk 1]
```diff
@@ -106,6 +106,8 @@ v0.38.0-rc.0 (2023-11-16)
 
 - Updated windows exporter to use prometheus-community/windows_exporter commit 1836cd1. (@mattdurham)
 
+- Added support to convert resource attributes to Prometheus-compatible datapoint attributes in `otelcol.exporter.prometheus`. (@hainenber)
```
Comment (ptodev) [inline CHANGELOG.md:109]:
```suggestion
- Added support to convert resource attributes to Prometheus labels in `otelcol.exporter.prometheus`. (@hainenber)
```
I think "attribute" and "datapoint" are OTel terms, so probably it's good enough just to say its a Prometheus label.

File: component/otelcol/exporter/prometheus/internal/convert/convert_test.go [hunk 2]
```diff
@@ -838,6 +839,233 @@ func TestConverter(t *testing.T) {
...
+						}]
+					}]
+				}]
+			}`,
+			expect: `
+				# TYPE test_metric_gauge gauge
+				test_metric_gauge{bar_one="bar",foo_one="foo",instance="instance",service_instance_id="instance",job="myservice",service_name="myservice",raw="test"} 1234.56
+			`,
+			enableOpenMetrics:             true,
+			resourceToTelemetryConversion: true,
```
Comment (ptodev) [inline component/otelcol/exporter/prometheus/internal/convert/convert_test.go:881]: It'd be nice to have a test or two where we have resource attributes but `resourceToTelemetryConversion: false`

File: docs/sources/flow/reference/components/otelcol.exporter.prometheus.md [hunk 3]
```diff
@@ -46,6 +46,7 @@ Name | Type | Description                                               | Defaul
 `add_metric_suffixes` | `boolean` | Whether to add type and unit suffixes to metrics names.   | `true` | no
 `gc_frequency` | `duration` | How often to clean up stale metrics from memory.          | `"5m"` | no
 `forward_to` | `list(receiver)` | Where to forward converted Prometheus metrics.            | | yes
+`resource_to_telemetry_conversion` | `boolean` | Whether to convert OTel resource attributes to Prometheus-compatible datapoint attributes | `false` | no
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.exporter.prometheus.md:49]:
```suggestion
`resource_to_telemetry_conversion` | `boolean` | Whether to convert OTel resource attributes to Prometheus labels. | `false` | no
```

**Review by ptodev — APPROVED** [review]
Thank you! LGTM - I just added a few minor suggestions.

File: component/otelcol/exporter/prometheus/internal/convert/convert_test.go (hunk 2 above)
Comment (hainenber) [reply component/otelcol/exporter/prometheus/internal/convert/convert_test.go:881]: Added in latest commit. PTAL. Thanks!

**Pull request merged**